### DISCUSSION TITLE
implicitly typed comments

### DISCUSSION THREAD

**Discussion by lazyloader** [dotnet/csharplang#1651] [Language Ideas] [answered]
if i did this code

```csharp
and if i wrote some comment up here that describes foo
public void foo()
    {
    ...
    }
```

This doesn't compile, which is completely frustrating. I have to prefice it with `//` for it to work. 

How about, instead, the compiler detects that that was meant to be a comment so i don't have to type `//`?

**Comment by CyrusNajmabadi** [comment]
> How about, instead, the compiler detects that that was meant to be a comment 

How would the compiler do that?  Can you be more specific about how you would intend this to work?

**Comment by stepanbenes** [comment]
He is just trolling...

**Comment by lazyloader** [comment]
@stepanbenes I am not, I am just trying to share ideas on how to make the language shorter.

@CyrusNajmabadi It would work by when it finds a string of invalid identifies (longer than say 5, but the exact number could be a setting that the user can change in Visual Studio) and it is placed ontop of a method, class, `if` block, or inlined at the end of a statement or any other place where we tend to put comments, the compiler will assume it's a comment. 

There could also be more settings on how changing where we want implicit comments. Maybe only at the top of methods. 

It would highlight it correctly as if it were a normal comment.

**Comment by CyrusNajmabadi** [comment, marked as answer]
@lazyloader Instead of proposing such a drastic (and contentious) language change.  I would suggest instead just investing in writing an analyzer to look for this, and to offer the fix you want.  

> @CyrusNajmabadi It would work by when it finds a string of invalid identifies (longer than say 5, but the exact number could be a setting that the user can change in Visual Studio) and it is placed ontop of a method, class, if block, or inlined at the end of a statement or any other place where we tend to put comments, the compiler will assume it's a comment.

What happens if hte user uses non-identifiers?  For example, something like "if" is a keyword.  What if they use a number?  What if they use punctuation?  Can they end their sentences with a period?  Or a question mark?  etc. etc.

**Comment by HaloFour** [comment]
@lazyloader

> , I am just trying to share ideas on how to make the language shorter.

Shorter does not mean better.  In this case, doubly so.  It's two characters, which virtually all IDEs will add for you via keyboard shortcut or button click.

**Comment by stepanbenes** [comment]
@lazyloader You want to make the language shorter? It is the reason why you proposed #1619?

**Comment by theunrepentantgeek** [comment]
Here's an example of why this is a bad idea ...

Imagine you work in a team with coding standards that require method modifiers to be on the line prior to the method name, like this:

```
protected virtual
void MagicMethod(string incantation)
{
    //
}
```

Note: ***not*** a style layout I personally endorse, but one I have encountered.

Now someone makes a simple spelling error modifying the visibility of the method:

```
pirvate protected virtual
void MagicMethod(string incantation)
{
    //
}
```

Instead of a compilation error, the first line is silently treated as a comment and the default visibility of *private* is applied.

If no other code depended on the method being non-private or virtual, things would silently compile and appear to work - but with the wrong semantics.

Worse, if there **was** code that depended on the modification, any errors reported by the compiler would appear a long way away from the actual cause of the problem.

**Comment by DavidArno** [comment]
If you want to make your code shorter, then give `foo` a meaningful name, which describes what it does, removing the need for a comment in the first place.

**Comment by Unknown6656** [comment]
**Hey**, guys, I have an Idea:
How about we add a feature to the compiler, which automatically recognises what code we would _want_ to write -- and the compiler writes and compiles it for us.
It could _drastically_ reduce the code size, e.g.:

- A "Hello, World!"-application could now look like this:
```csharp
```

- And here I have an application which proves that `P = NP`:
```csharp
```

<hr/>
:trollface:™

**Comment by bondsbw** [comment]
@Unknown6656 Unfortunately the same code behaves differently depending on how much you remember the next day.

**Comment by Unknown6656** [comment]
@bondsbw A.k.a. the too-much-beer-in-the-evening-effect :wink:

**Comment by lazyloader** [comment]
@stepanbenes  shorter but not so short that it makes it harder to read and understand. `if (isTrue) do stuff` isn't as easy to read as `if (isTrue) then do stuff` so it's worth having to type a little bit more.

**Comment by iam3yal** [comment]
@lazyloader, You're using the wrong kind of comment, document types and members should be done through XML comments as opposed to regular comments whereas the latter should be used to either **temporarily** comment code or provide an explanation to things that you can't express in code like providing a reason for a piece of code to exist the way it is but anything between these two ends should really be explained by the code itself.

But really, if writing `//` causes frustration then I'd argue that you write too many comments whereas you should improve the code so you wouldn't need to write them at all, the language should not help you write even more comments and if you write few, you're just exaggerating. 😄
